shell/platform/linux/fl_scrolling_manager.h

```diff
@@ -0,0 +1,113 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_SCROLLING_MANAGER_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_SCROLLING_MANAGER_H_
+
+#include <gdk/gdk.h>
+#include <functional>
+
+#include "flutter/shell/platform/linux/fl_scrolling_view_delegate.h"
+
+G_BEGIN_DECLS
+
+#define FL_TYPE_SCROLLING_MANAGER fl_scrolling_manager_get_type()
+G_DECLARE_FINAL_TYPE(FlScrollingManager,
+                     fl_scrolling_manager,
+                     FL,
+                     SCROLLING_MANAGER,
+                     GObject);
+
+/**
+ * fl_scrolling_manager_new:
+ * @view_delegate: An interface that the manager requires to communicate with
+ * the platform. Usually implemented by FlView.
+ *
+ * Create a new #FlScrollingManager.
+ *
+ * Returns: a new #FlScrollingManager.
+ */
+FlScrollingManager* fl_scrolling_manager_new(
+    FlScrollingViewDelegate* view_delegate);
+
+/**
+ * fl_scrolling_manager_set_last_mouse_position:
+ * @manager: the #FlScrollingManager self.
+ * @x: the mouse x-position, in window coordinates.
+ * @y: the mouse y-position, in window coordinates.
+ *
+ * Inform the scrolling manager of the mouse position.
+ * This position will be used when sending scroll pointer events.
+ */
+void fl_scrolling_manager_set_last_mouse_position(FlScrollingManager* self,
```

Comment:
Replace `self` with `manager` to match the comments above. (and other cases below).